app/controllers/foreman_bootdisk/disks_controller.rb

```diff
@@ -81,5 +86,13 @@ def error_rendering(exception)
       error("#{msg}: #{exception.message}")
       ::Foreman::Logging.exception(msg, exception)
     end
+
+    def prolong_token(host)
+      return unless host.token
```

Comment:
@lzap please keep me honest, I think host can be in build mode but token is nil, because it has expired and afterwards cleaned up (there's some cleanup mechanism for expired tokens). Perhaps the check should be that host is in build mode, if it is, either prolong an existing token or create a new one?

I'm not sure about this, so @domitea please investigate the token expiration/cleanup behavior and share your findings here.

Reply:
I've checked the code of Token (https://github.com/theforeman/foreman/blob/develop/app/models/concerns/hostext/token.rb) and I don't think that's a problem because when the host is in build mode, it has a token as far as I understand the code.

Further explanation:

There is a `expire_token` method at Token. This method is only used at `refresh_token_on_build` that is called every time before validation of token (https://github.com/theforeman/foreman/blob/develop/app/models/concerns/hostext/token.rb#L11). And when you at the code to the refresh one, you can see that when there is a creating a new token when the host become in build mode. This applies to the new hosts and even for the old one when they are changing mode. 

What do you think, @lzap and @ares ?

Reply:
> there's some cleanup mechanism for expired tokens

There actually isn't. Tokens are only generated or deleted when a host enters/exits build mode. There is not expiration outside of that AFAIK, unattended controller checks for expiration time and if it is expired it is treated as non-existing that's all.

So I think the line here should be:

```suggestion
      return if !host.token_expired? || Setting[:token_duration] == 0
```

I would suggest not to prolong tokens on each download, this could have some security implications. Lets prolong it only when it expired.

Reply:
I think it's a good idea @lzap! 
However there is still some little probability that user will download image just before (let's say couple of minutes) token expiration. I know that a really edge case and the probability is very small but It's there.

It's enough solution for @ares?

Reply:
On a second thought I think if an attacker have ability to download bootdisk iso, then there is no harm in prolonging it on every download. It can be little confusing tho so I agree with prolonging it on every download, let's put a sentence on the Help screen for host bootdisks: Each download will renew provisioning token expiration when in use. 

Therefore I am expecting something like this:

    return if Setting[:token_duration] == 0 || host.token.nil?

Reply:
I agree and would prefer to prolong on every download. Each newly downloaded bootdisk should have the full provisioning window.